### PR TITLE
Assignment9

### PR DESCRIPTION
**Pull request by RichardTeller**
Unfortunately a little late to my PR.

Any feedback is still appreciated.

### CODE REVIEW

File: SecretSanta/test/SecretSanta.Web.UITests/AddUsersPage.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        public IWebElement SubmitButton =>
+            Driver
+                .FindElements(By.CssSelector("button.is-primary"))
+                .Single(x => x.Text == "Submit");
+
+        public IWebElement CancelLink => Driver.FindElement(By.LinkText("Cancel"));
+
+        public AddUsersPage(IWebDriver driver)
+        {
+            Driver = driver ?? throw new ArgumentNullException(nameof(driver));
```
Comment (kamonson) [inline SecretSanta/test/SecretSanta.Web.UITests/AddUsersPage.cs]: Typically if you throw an exception you should handle it in the same location.  If the driver is null, catch the exception and spin up a "default" driver?

File: SecretSanta/test/SecretSanta.Web.UITests/AddUsersPage.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Linq;
+using OpenQA.Selenium;
+
+namespace SecretSanta.Web.UITests
+{
+    public class AddUsersPage
+    {
+        public const string Slug = UsersPage.Slug + "/Add";
+        public IWebDriver Driver { get; }
```
Comment (kamonson) [inline SecretSanta/test/SecretSanta.Web.UITests/AddUsersPage.cs]: Lines 10 - 24 seem to be duplicated in other classes.  It is a much better idea to create a base class and inherit from that so you don't duplicate these function body members

File: SecretSanta/test/SecretSanta.Web.UITests/HomePageTests.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace SecretSanta.Web.UITests
+{
+    [TestClass]
+    public class HomePageTests
+    {
+        private const string RootUrl = "https://localhost:44398/";
```
Comment (kamonson) [inline SecretSanta/test/SecretSanta.Web.UITests/HomePageTests.cs]: Consider moving this into the "base class" mentioned earlier

File: SecretSanta/test/SecretSanta.Web.UITests/UsersPage.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+using System.Text;
+using OpenQA.Selenium;
+
+namespace SecretSanta.Web.UITests
+{
+    public class UsersPage
+    {
+        public IWebDriver Driver { get; set; }
+        public const string Slug = "Users";
+        public IWebElement HomeLink => Driver.FindElement(By.CssSelector(".navbar-brand>a"));
```
Comment (kamonson) [inline SecretSanta/test/SecretSanta.Web.UITests/UsersPage.cs]: Mentioned above

File: SecretSanta/test/SecretSanta.Web.UITests/UsersPageTests.cs [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace SecretSanta.Web.UITests
+{
+    [TestClass]
+    public class UsersPageTests
+    {
+        private const string RootUrl = "https://localhost:44398/";
```
Comment (kamonson) [inline SecretSanta/test/SecretSanta.Web.UITests/UsersPageTests.cs]: Consider moving to base class

File: SecretSanta/test/SecretSanta.Web.UITests/AddUsersPage.cs (hunk 1 above)
Comment (Keboo) [reply SecretSanta/test/SecretSanta.Web.UITests/AddUsersPage.cs]: I would tend to disagree with @kamonson here. Some exceptions are not expected to be handled by your code, things like `NullReferenceException` and `ArgumentException` (which includes the derived `ArgumentNullException`) are typically used to denote programmer error and not expected to be handled.

File: SecretSanta/test/SecretSanta.Web.UITests/UsersPage.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using OpenQA.Selenium;
+
+namespace SecretSanta.Web.UITests
+{
+    public class UsersPage
+    {
+        public IWebDriver Driver { get; set; }
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Web.UITests/UsersPage.cs]: This can be a readonly auto property

**Pull request closed without merging**